### DISCUSSION TITLE
Comments?

### DISCUSSION THREAD

**Discussion by loic-joly-sonarsource** [microsoft/ifc-spec#42] [General]
Many tools have some need to represent some information that has no semantic value for a program:

1. Name of local variable
2. Comments
3. The fact that a piece of code is the result of a macro expansion (or of full macro-metaprogramming)
4. The fact that a type was mentioned directly or through an alias
5. Source locations of nodes
...

As far as I understand, some of those are already represented in ifc (1, 5), some of them are not, but could easily fit in the model (4), some of them are currently totally out of scope (2, 3)

Is the ultimate goal to represent all of them, or is the purpose of ifc to only focus on semantically meaningful data?

**Comment by GabrielDosReis** [comment]
> Is the ultimate goal to represent all of them, or is the purpose of ifc to only focus on semantically meaningful data?

Ideally, all of this should be represented in the IFC files, some by default and others possibly through a switch.  For example, the MSVC team is working towards a rich source location data structure that enables (3).  Point (2) is point of tension between keeping only semantically-relevant information (so as to minimize recompilation of dependent components) and being 100% faithful to the lexical content of the input source file (which would suffer from drawback similar to input source files in terms of changes, but would allow storing interface documentations with their declarations).  It is ,most likely a candidate for a compiler option.
